### DISCUSSION TITLE
Add/Remove elements to an existing instance

### DISCUSSION THREAD

**Discussion by gDuval60** [fancyapps/ui#470] [General]
Hello ,

Is it possible to add or remove an element in an existing Fancybox instance?

I have a page with several albums. When I click on an album, I open a Fancybox instance in which I have added custom buttons to add and remove images but I can't find a native function allowing me to do that.

THANKS.

**Comment by fancyapps** [comment]
Hi, What do you mean by "elements"? Fancybox is build around Carousel component. So, you want to add something in the current slide of the Carousel, or you want to add/remove slides?

**Reply by gDuval60** [reply]
I'm talking about slides that can contain photos and videos

**Reply by fancyapps** [reply]
Get reference to Carousel instance and then use API, like this:

```js
Fancybox.getInstance().carousel.appendSlide({ src : "https://lipsum.app/id/1/800x600", type: "image"}); 
```

**Reply by gDuval60** [reply]
Thanks !!! This is exactly what I was looking for .

**Reply by gDuval60** [reply]
Hi, How should a thumbnail be integrated into the appendSlide function ?

**Reply by gDuval60** [reply]
?

**Reply by fancyapps** [reply]
Sorry, you can use `thumb` key to set the thumbnail source, but the thumbnail bar won't update automatically. I will try to implement this.

**Reply by gDuval60** [reply]
Hello,

Were you able to integrate the automatic update of the thumbnail bar ?

**Reply by fancyapps** [reply]
@gDuval60 I'm actually working on it right now. So please stay tuned.

**Reply by gDuval60** [reply]
Ok perfect ! I'm looking forward to it

**Reply by fancyapps** [reply]
See https://github.com/fancyapps/ui/issues/71#issuecomment-1793679729
